crates/iceberg-catalog/src/catalog/views/commit.rs

```diff
@@ -81,56 +86,137 @@ pub(crate) async fn commit_view<C: Catalog, A: Authorizer + Clone, S: SecretStor
             None,
         ))?;
 
-    let GetWarehouseResponse {
-        id: _,
-        name: _,
-        project_id: _,
-        storage_profile,
-        storage_secret_id,
-        status,
-        tabular_delete_profile: _,
-    } = C::require_warehouse(warehouse_id, t.transaction()).await?;
-    require_active_warehouse(status)?;
+    let warehouse = C::require_warehouse(warehouse_id, t.transaction()).await?;
+    let storage_profile = &warehouse.storage_profile;
+    let storage_secret_id = warehouse.storage_secret_id;
+    require_active_warehouse(warehouse.status)?;
+    t.commit().await?;
 
+    // Verify assertions (only needed once)
     check_asserts(requirements.as_ref(), view_id)?;
 
+    // Prepare for event publishing (serialize body once)
+    let body = maybe_body_to_json(&request);
+
+    // Start the retry loop
+    let mut attempt = 0;
+    loop {
```

Comment:
I'm wondering if we can reduce nesting here a bit if we'd use tryhard which we already use within the io module

Reply:
(same at the table loop)